src/drivers/net/lan91c111/lan91c111.c

```diff
@@ -252,89 +204,127 @@ static const struct net_driver lan91c111_drv_ops = {
 	.set_macaddr = lan91c111_set_macaddr
 };
 
-static irq_return_t lan91c111_int_handler(unsigned int irq_num, void *dev_id) {
+static int lan91c111_tx_pend(struct net_device *dev) {
+	struct sk_buff *skb;
+	struct smc_local *smc;
+	int ret;
+
+	smc = netdev_priv(dev);
+
+	skb = skb_queue_pop(&smc->pending_queue);
+	if (!skb) {
+		return 0;
+	}
+	lan91c111_send_hw(skb);
```

Comment:
Looks like theese lines can be removed as `while` look does the same thing.

Reply:
Yes, but, when we first-time appear in this point it means that ALLOC_TX command was finished. Of cause we can call ALLOC_TX again, it's not significant I think